File: gpstracker/src/main/java/com/golriz/gpstracker/TrackingServiceRestartReceiver.java
package com.golriz.gpstracker;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

public class TrackingServiceRestartReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent.getAction() != null) {
            if (LocationTrackerService.stopTime) {
                LocationTrackerService.stopTime = false;
                return;
            }
            if (intent.getAction().equals("service.locationTracker.stopped")) {

                
                context.startService(new Intent(context, LocationTrackerService.class));
            }

        }
    }
}

File: gpstracker/src/main/java/com/golriz/gpstracker/Core/TrackingServiceRestartReceiver.kt
package com.golriz.gpspointer.Config

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class TrackingServiceRestartReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action != null) {
            if (LocationService.stopTime) {
                LocationService.stopTime = false
                return
            }
            if (intent.action == "service.locationTracker.stopped") {


                LocationTracker("my.action", null)
                    .setInterval(5000)
                    .setGps(true)
                    .setNetWork(true)
                    .start(context)
            }

        }
    }
}